### PR TITLE
Bootstrap 4 masthead navbar menu responsiveness

### LINKED ISSUE #5393: Bootstrap header navbar menu is not fully responsive
### Descriptive summary

Header navbar menu does not render as expected in smaller viewports. The navbar toggler does not align in a row-like behavior and menu items are centered horizontal when the menu is expanded.

image

image

image

### Expected behavior

When responsive to smaller viewports:

- [ ] Navbar header has flex direction of row
- [ ] Navbar Toggler aligns right
- [ ] Navbar items align left when expanded
- [ ] Navbar child menu items expand to the width of their parent

### PR DESCRIPTION
**Pull request by mathewjordan**
Fixes #5393  ; refs #5276 

## Summary

This resolves some issues introduced in the upgrade to Bootstrap 4 and utilizes new flexbox utility classes to improve alignment behaviors.

## Changes proposed in this pull request:
- Refines flexbox behavior for responsiveness
- Add `placement_class` (is this an argument?) on masthead render and removes CSS overrides
- Updates locale picker dropdown structure and converts unordered list to div/anchors

### CODE REVIEW

**Review by adamjarling — APPROVED** [review]
Way better!

File: app/assets/stylesheets/hyrax/_styles.scss [hunk 1]
```diff
@@ -63,16 +63,6 @@ label.disabled {
   display: flex;
 }
 
-.navbar {
```
Comment (adamjarling) [inline app/assets/stylesheets/hyrax/_styles.scss:66]: Nice to see this stuff get thinned out.

File: app/assets/stylesheets/hyrax/dashboard.scss [hunk 2]
```diff
@@ -74,12 +74,6 @@ body.dashboard {
    * replace the "navbar-static-top" class with "navbar-fixed-top" for the Admin UI only
    */
   #masthead {
-    border: 0;
-    left: 0;
-    position: fixed;
-    right: 0;
-    top: 0;
-
```
Comment (adamjarling) [inline app/assets/stylesheets/hyrax/dashboard.scss:82]: I fought with removing and adding this a few times.  Glad to see you got the `fixed-top` implementation going.

**Review by adamjarling — CHANGES_REQUESTED** [review]
Actually @mathewjordan Can you take a look at this on the Dashboard page?   This is what I was fighting with when working on the Sidebar issue yesterday.

image

The sidebar tucks underneath the fixed-top nav bar.

**Comment by mathewjordan** [comment]
> 

Oh yeah @adamjarling. Likely something in that sidebar that is being too explicit. I'll look at that.

**Comment by mathewjordan** [comment]
@adamjarling This should be good now.

**Review by adamjarling — APPROVED** [review]
Lookin good.

**Pull request merged**